### PR TITLE
do dispose async

### PR DESCRIPTION
**Pull request by SimonCropp**
i assume you would prefer to do dispose in an async manner where possible?

### CODE REVIEW

File: src/Microsoft.Tye.Core/ProcessUtil.cs [hunk 1]
```diff
@@ -109,7 +109,7 @@ public static class ProcessUtil
             process.BeginErrorReadLine();
 
             var cancelledTcs = new TaskCompletionSource<object?>();
-            using var _ = cancellationToken.Register(() => cancelledTcs.TrySetResult(null));
+            await using var _ = cancellationToken.Register(() => cancelledTcs.TrySetResult(null));
```
Comment (rynowak) [inline src/Microsoft.Tye.Core/ProcessUtil.cs]: Wow. I learned something new today thanks to this PR!

@davidfowl did you know about this? https://github.com/dotnet/runtime/blob/master/src/libraries/System.Private.CoreLib/src/System/Threading/CancellationTokenRegistration.cs#L48

**Comment by rynowak** [comment]
CI failures look unrelated to your changes. Rerunning.

File: src/Microsoft.Tye.Core/ProcessUtil.cs (hunk 1 above)
Comment (SimonCropp) [reply src/Microsoft.Tye.Core/ProcessUtil.cs]: do u want a PR that does CancellationTokens?

Comment (davidfowl) [reply src/Microsoft.Tye.Core/ProcessUtil.cs]: Yah

**Comment by davidfowl** [comment]
There’s a machine in the ci pool with busted dev certs

**Comment by rynowak** [comment]
> There’s a machine in the ci pool with busted dev certs 

Is there a way to repair it? We could run the dev cert tool as part of our CI.

File: src/Microsoft.Tye.Core/ProcessUtil.cs (hunk 1 above)
Comment (SimonCropp) [reply src/Microsoft.Tye.Core/ProcessUtil.cs]: name for CancellationToken variables? my preference is `cancellation`, but you already have some with `cancellationToken`

**Comment by davidfowl** [comment]
I’m going to do that the next time I can repro it

**Pull request merged**